tests/sles4sap/monitoring_services.pm

```diff
@@ -186,6 +186,40 @@ sub configure_sap_host_exporter {
     upload_logs $metrics_file;
 }
 
+sub configure_alloy {
+    # Install needed packages
+    zypper_call 'in alloy system-user-alloy';
+
+    # Collect some info
+    zypper_call 'info --provides alloy';
+    script_run 'rpm -qf $(which alloy)';
+    script_run 'rpm -ql $(rpm -qf $(which alloy))';
+
+    # Enable and verify the service
+    systemctl 'enable --now alloy';
+    systemctl 'status alloy';
+    systemctl 'is-active alloy';
+    assert_script_run 'journalctl -u alloy';
+
+    # Check the config files
+    my $config_file_line = script_output 'grep CONFIG_FILE /etc/sysconfig/alloy';
+    # The following regex extracts the file path from a line like 'CONFIG_FILE="/path/to/file"'.
+    # - ^CONFIG_FILE\s*=\s* : Matches the CONFIG_FILE variable at the beginning of the line, allowing for flexible whitespace.
+    # - (["'])?             : Capturing group 1. Optionally matches an opening quote (' or "). It handles optional single or double quotes around the path.
+    # - (.*?)               : Capturing group 2. Lazily captures the file path.
+    # - \1?                 : Optionally matches the same quote captured in group 1, ensuring quotes are paired. It also works in case of no quotes.
+    # The path will be available in group $2
+    die 'Could not determine config file path from /etc/sysconfig/alloy.' unless ($config_file_line && $config_file_line =~ /^CONFIG_FILE\s*=\s*(["'])?(.*?)\1?$/);
```

Comment:
only suggestion, maybe it doesn't work.
`/^CONFIG_FILE\s*=\s*(["'])?(.*?)\1?$/`  ==> `/^CONFIG_FILE\s*=\s*(["'])?(.*?)\1?($|\s*)/`

Reply:
I tested it but it does not work